File: code401challenges/src/main/java/code401challenges/RepeatedWord/RepeatedWord.java
package code401challenges.RepeatedWord;

import java.util.HashSet;

public class RepeatedWord {

    public static String repeatedWord (String words) {
        HashSet<String> map = new HashSet<>();

        String[] array = words.split(" ");

        for (String word : array) {
            String realWord = word.split(",")[0];
            if (map.contains(realWord.toLowerCase())) {
                return word;
            }
            map.add(realWord.toLowerCase());
        }
        return "";
    }
}

File: js401challenges/HashTable/RepeatedWord.js
'use strict';

function firstRepeatedWord (input) {

  let set = new Set();
  let words = input.split(' ');

  let repeatedWord = '';

  words.forEach( word => {
    let purifying = word.split(',')[0];
    let pureWord = purifying.split('.')[0].toLowerCase();

    if(set.has(pureWord)) {
      return repeatedWord;
    }

    set.add(pureWord);
  })

  return repeatedWord;

}

module.exports = firstRepeatedWord;
